spec.bs

```diff
@@ -5327,6 +5297,84 @@ A <dfn>per signals url bid generator</dfn> is an [=ordered map=] whose [=map/key
 representing [=interest group/joining origins=], and whose [=map/values=] are [=lists=] of
 [=interest groups=].
 
+<h3 id=trusted-bidder-signals-batcher>Trusted bidding signals batcher</h3>
+
+A <dfn>trusted bidding signals batcher</dfn> helps manage merging multiple trusted bidding signals
+into smaller number of fetches. It's a [=struct=] with the following  [=struct/items=]:
+
+<dl dfn-for="trusted bidding signals batcher">
+  : <dfn>all trusted bidding signals</dfn>
+  :: An [=ordered map=] whose [=map/keys=] are [=strings=], and [=map/values=] are [=strings=].
+     Contains all trusted bidding signals collected thus far.
+  : <dfn>data version</dfn>
+  :: An {{unsigned long}} or null, initially null. Data version number provided by the last fetch.
+  : <dfn>keys</dfn>
+  :: An [=ordered set=] of [=strings=]. Describes the keys collected to be fetched in the current
+     batch thus far.
+  : <dfn>ig names</dfn>
+  :: An [=ordered set=] of [=strings=]. Describes the interest group names to be fetched in the
+     current batch thus far.
+  : <dfn>length limit</dfn>
+  :: A {{long}}, initially the maximum value that it can hold. Describes the URL length limit
+     the current batch is limited to, the smallest of the limits of the interest groups included.
+</dl>
+
+<div algorithm>
+To <dfn>fetch the current outstanding trusted signals batch</dfn> given a
+[=trusted bidding signals batcher=] |trustedBiddingSignalsBatcher|, a [=URL=] |signalsUrl|, an
+{{unsigned short}}-or-null |experimentGroupId|, an [=origin=] |topLevelOrigin|,
+and a [=string=] |slotSizeQueryParam|:
+
+  1. Let |biddingSignalsUrl| be the result of [=building trusted bidding signals url=] with
+     |signalsUrl|, |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/keys=],
+     |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/ig names=],
+     |experimentGroupId|, |topLevelOrigin|, and |slotSizeQueryParam|.
+  1. Let « |partialTrustedBiddingSignals|, |dataVersion| » be the result of [=fetching trusted signals=]
+     with |biddingSignalsUrl| and true.
+  1. [=map/For each=] |key| → |value| in |partialTrustedBiddingSignals|:
+    1. Set |trustedBiddingSignalsBatcher|'s
+      [=trusted bidding signals batcher/all trusted bidding signals=][|key|] to |value|.
+  1. Set |trustedBiddingSignalsBatcher|'s [=trusted bidding signals batcher/data version=] to |dataVersion|.

```

Comment:
This should match what's before this CL, but it feels wrong to me?

Reply:
I feel like we should probably key it by name, so if data-version changes each call gets the right value?

Reply:
Yeah I believe this is the correct one. Set A to B means A = B.

Reply:
The issue isn't the assignment order. It's that it always uses the data-version of the last fetch, rather than of the fetch used for a particular IG.

Reply:
Didn't see your latest comment. Agreed that there should be a key for the data version to store different ones from different batch reqeusts.

Reply:
And done. PTAL.